src/seedu/addressbook/commands/EditCommand.java

```diff
@@ -0,0 +1,66 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.common.Messages;
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.*;
+import seedu.addressbook.data.tag.Tag;
+import seedu.addressbook.data.tag.UniqueTagList;
+import static seedu.addressbook.ui.TextUi.DISPLAYED_INDEX_OFFSET;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Adds a person to the address book.
+ */
+public class EditCommand extends Command {
+
+    public static final String COMMAND_WORD = "edit";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD +
+            ": Edits the person identified by the index number in the last shown person listing.\n"
+            + "Parameters: INDEX NAME [p]p/PHONE [p]e/EMAIL [p]a/ADDRESS [t/tag]...\n"
+            + "Example: " + COMMAND_WORD
+            + " 1 John Doe p/98765432 e/johnd@gmail.com a/311, Clementi Ave 2";
+
+    public static final String MESSAGE_EDIT_PERSON_SUCCESS = "Person edited: %1$s";
+
+    private int targetIndex;
+    private final Person toEdit;
+
+    /**
+     * Convenience constructor using raw values.
+     *
+     * @throws IllegalValueException if any of the raw values are invalid
+     */
```

Comment:
constructor creates a person object from the parameters given.
This newly created person then replaces the person stored at the target index during execute().